src/ducks/categorization/services.js

```diff
@@ -186,6 +204,37 @@ const getLocalClassifierOptions = transactionsWithManualCat => {
   }
 }
 
+export const atomicReweight = (classifier, category, word, frequencyCount) => {
+  const newFrequencyCount = 1 + Math.log(frequencyCount)
+  const deltaFrequencyCount = frequencyCount - newFrequencyCount
+  // update the right entries of the classifier's parameters
+  classifier.vocabulary[word] -= deltaFrequencyCount
+  classifier.wordCount[category] -= deltaFrequencyCount
+  classifier.wordFrequencyCount[category][word] = newFrequencyCount
+}
+
+export const reweightModel = classifier => {
+  // loop over categories in the wordFrequencyCat attribute
+  const wordFrequencyCount = classifier.wordFrequencyCount
+  // for each category
+  for (const category in wordFrequencyCount) {
+    if (wordFrequencyCount.hasOwnProperty(category)) {
+      // extract its word-frequency count `wfc`
+      const categoryWordsFrequencyClounts = wordFrequencyCount[category];
```

Comment:
There's a typo I think: `categoryWordsFrequencyClounts` => `categoryWordsFrequencyCounts`. No bug because you use the same name after, but maybe you can rename the variable.

Reply:
The habbit of typing "(Cozy) Cloud" ;) Thanks